pandas/tests/copy_view/test_core_functionalities.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import pytest
+
+from pandas import DataFrame
+import pandas._testing as tm
+from pandas.tests.copy_view.util import get_array
+
+
+def test_assigning_to_same_variable_removes_references(using_copy_on_write):
+    df = DataFrame({"a": [1, 2, 3]})
+    df = df.reset_index()
+    arr = get_array(df, "a")
+    df.iloc[0, 1] = 100  # Write into a
+
+    assert np.shares_memory(arr, get_array(df, "a"))
+
+
+def test_setitem_dont_track_unnecessary_references(using_copy_on_write):
+    df = DataFrame({"a": [1, 2, 3], "b": 1, "c": 1})
+
+    df["b"] = 100
+    arr = get_array(df, "a")
+    df.iloc[0, 0] = 100  # Check that we correctly removed the reference
+    assert np.shares_memory(arr, get_array(df, "a"))
```

Comment:
Ah, __setitem__ splits the blocks under the hood to avoid calling np.delete(). This is basically what we are also testing in the del df["a"] test above.

I want to make sure that the spit blocks don't have any references on them if the parent block did not have any reference. Parent block went out of scope, so the new blocks should be clear of references as well.

Reply:
Ah, OK, thanks for the explanation, wasn't thinking about the side effect of the non-inplace `__setitem__` of splitting the block